app/src/main/java/com/example/cristiana/workshop1/MainActivity.java

```diff
@@ -0,0 +1,89 @@
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.cristiana.workshop1.model.GitHub;
+import com.example.cristiana.workshop1.model.LoginData;
+
+import org.w3c.dom.Text;
+
+import okhttp3.Credentials;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+
+    TextView mUsername, mPassword;
+
+    private static final String TAG = "MainActivity";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        /* preferences */
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        if (preferences.getBoolean("loggedIn", false)) {
+            Intent intent = new Intent(this, ProfileActivity.class);
+            startActivity(intent);
+            finish();
+            return;
+        }
+
+        mUsername = (TextView) findViewById(R.id.username);
+        mPassword = (TextView) findViewById(R.id.password);
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().hide();
+        }
+
+        Button loginButton = (Button) findViewById(R.id.LoginButton);
+        loginButton.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        if (v.getId() == R.id.LoginButton) {
+            performLogin(mUsername.getText().toString(), mPassword.getText().toString());
+        }
+    }
+
+
+    private void performLogin(String username, String password) {
+        // Make a network call and authenticate the user
+        Call<LoginData> callable = GitHub.Service.Get().checkAuth(Credentials.basic(username, password));
+
+        callable.enqueue(new Callback<LoginData>() {
+            @Override
+            public void onResponse(Call<LoginData> call, Response<LoginData> response) {
+                if (response.isSuccessful()) {
+                    SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(MainActivity.this);
+                    preferences.edit().putBoolean("loggedIn", true).apply();
+
+                    Intent intent = new Intent(MainActivity.this, ProfileActivity.class);
+                    startActivity(intent);
+                    finish();
+                } else {
+                    Toast.makeText(MainActivity.this, "An error ocurred!", Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<LoginData> call, Throwable t) {
+                t.printStackTrace();
```

Comment:
Filtreaza te rog dupa erorile care pot fi prinse in onFailure (nu sunt toate dubioase :P), e.g.:
* timeout: TimeoutException
* lipsa conexiune internet: UnknowHostException
etc